.github/workflows/integrationTest.yml

```diff
@@ -14,18 +14,73 @@ env:
   PASSPHRASE: ${{ secrets.PASSPHRASE }}
   GPG_KEY_NAME: ${{ secrets.GPG_KEY_NAME }}
   GPG_TTY: $(tty)
+  ECR_INTEGRATION_TEST_REPO: "cwagent-integration-test"
 
 on:
   push:
     branches:
       - master
+    
   workflow_dispatch:
 
 concurrency:
   group: ${{ github.workflow }}-${{ github.ref_name }}
   cancel-in-progress: true
 
 jobs:
+  MakeDockerImage:
+    name: 'MakeDockerImage'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Cache if success
+        id: build-docker-image
+        uses: actions/cache@v2
+        with:
+          path: |
+            RELEASE_NOTES
+          key: build-docker-image-${{ github.run_id }}
+
+      - name: Configure AWS Credentials
+        if: steps.build-docker-image.outputs.cache-hit != 'true'
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.TERRAFORM_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.TERRAFORM_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-west-2
+
+      - name: Login ECR
+        if: steps.build-docker-image.outputs.cache-hit != 'true'
+        id: login-ecr
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Set up Docker Buildx
+        if: steps.build-docker-image.outputs.cache-hit != 'true'
+        uses: docker/setup-buildx-action@v1
+
+      - name: Set up QEMU
+        if: steps.build-docker-image.outputs.cache-hit != 'true'
+        uses: docker/setup-qemu-action@v1
+
+        #Build the cloudwatch agent image for two primary reasons:
+        #-Using the cloudwatch agent image to do the integration test (can be used for internal)
+        #-Export it for the future use in CD release pipeline
+        #Documentation: https://github.com/docker/build-push-action
+      - name: Build Cloudwatch Agent Image
```

Comment:
Not sure this works with the docker file since it curls the latest. It does not really build the deb from source under the hood. I think we need to talk this over. I don't think it does what it says.

Reply:
Since we can't merge this yet just blame it on me in the stand up saying that we dove deep and the current implementation does not match the docs.

Reply:
The Dockerfile which curls from the latest binary is [this one](https://github.com/aws/amazon-cloudwatch-agent/blob/master/amazon-cloudwatch-container-insights/cloudwatch-agent-dockerfile/Dockerfile). This one use `build-for-docker-${TARGETARCH}` so no worries on that though.

Reply:
Ignore this commit.